### PR TITLE
Terminate call if call view presentation is delayed.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/call/CallService.swift [hunk 1]
```diff
@@ -144,8 +144,10 @@ protocol CallServiceObserver: class {
                 if let oldValue = oldValue {
                     DeviceSleepManager.sharedInstance.removeBlock(blockObject: oldValue)
                 }
+                stopAnyActiveCallTimer()
                 if let call = call {
                     DeviceSleepManager.sharedInstance.addBlock(blockObject: call)
+                    self.startActiveCallTimer()
```
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: I feel like this slightly more strongly ensure the invariant "there is a timer IFF there is a call."

File: Signal/src/call/CallService.swift [hunk 2]
```diff
@@ -1574,9 +1573,8 @@ protocol CallServiceObserver: class {
     // MARK: CallViewController Timer
 
     var activeCallTimer: Timer?
```
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: We could rename this to `callTimer`.

File: Signal/src/call/CallService.swift (hunk 1 above)
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: Sounds reasonable.  We might want to rename some of the associated methods to give a hint as to their new roles.

maybe: 
- `startCallTimer` 
- `stopAnyCallTimer`
- `ensureUndisplayedCallTerminated`

Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: Renamed.

**Pull request merged**